qfieldsync/core/cloud_api.py

```diff
@@ -114,13 +122,69 @@ def from_reply(reply: QNetworkReply) -> Optional[CloudException]:
     return CloudException(reply, Exception(message))
 
 
+class CloudAuthMethod(Enum):
+    NONE = 0
+    CREDENTIALS = 1
+    SSO = 2
+
+
+def build_oauth2_auth_config_from_cloud_method(
+    auth_data: dict, related_uri: str, config_name: str = "qfieldcloud-sso"
```

Comment:
Is the `qfieldcloud-sso` a string under our control? If yes, the whole QField* codebase has a strong preference on `_` over `-`.